core/iwasm/fast-jit/fe/jit_emit_function.c

```diff
@@ -341,23 +336,34 @@ jit_compile_op_call_indirect(JitCompContext *cc, uint32 type_idx,
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
     /* Set native_ret to x86::eax */
     native_ret = jit_codegen_get_hreg_by_name("eax");
+
+    edx_hreg = jit_codegen_get_hreg_by_name("edx");
+    GEN_INSN(MOV, edx_hreg, elem_idx);
+    elem_idx = edx_hreg;
+
+    r9_hreg = jit_codegen_get_hreg_by_name("r9");
+    GEN_INSN(MOV, r9_hreg, argv);
+    argv = r9_hreg;
 #else
     native_ret = jit_cc_new_reg_I32(cc);
 #endif
 
     insn = GEN_INSN(CALLNATIVE, native_ret,
-                    NEW_CONST(PTR, (uintptr_t)wasm_call_indirect), 5);
+                    NEW_CONST(PTR, (uintptr_t)jit_call_indirect), 6);
     if (!insn) {
         goto fail;
     }
 
     *(jit_insn_opndv(insn, 2)) = cc->exec_env_reg;
     *(jit_insn_opndv(insn, 3)) = NEW_CONST(I32, tbl_idx);
-    *(jit_insn_opndv(insn, 4)) = element_indices;
-    *(jit_insn_opndv(insn, 5)) = NEW_CONST(I32, func_type->param_count);
-    *(jit_insn_opndv(insn, 6)) = argv;
+    *(jit_insn_opndv(insn, 4)) = elem_idx;
+    *(jit_insn_opndv(insn, 5)) = NEW_CONST(I32, type_idx);
+    *(jit_insn_opndv(insn, 6)) = NEW_CONST(I32, func_type->param_cell_num);
+    *(jit_insn_opndv(insn, 7)) = argv;
 
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
     jit_lock_reg_in_insn(cc, insn, native_ret);
```

Comment:
shall we lock `edx` and `r9' here?

Reply:
second thought, if `edx` and `r9` are "caller saved register", there is no need to lock them, right?

Or in other words, we should not lock any "caller saved register"